app/[locale]/sitemap.xml/route.ts

```diff
@@ -0,0 +1,69 @@
+import { NextRequest, NextResponse } from "next/server";
+import { XMLBuilder } from "fast-xml-parser";
+import {
+  generateSitemapUrl,
+  getSitemapData,
+  getSiteMapNewsData,
+} from "@/lib/api/sitemap";
+
+export async function GET(
+  request: NextRequest,
+  { params: { locale } }: LocaleProps
+) {
+  const pages = await getSitemapData(locale);
+  const pageData = pages.map(({ uri, dateUpdated }) => {
+    return {
+      loc: generateSitemapUrl(uri, locale),
+      lastmod: dateUpdated,
+    };
+  });
+  const { siteTitle, news } = await getSiteMapNewsData(locale);
+
+  const today = new Date();
+  const recentNewsThreshold = new Date(
+    today.getTime() - 1000 * 60 * 60 * 24 * 2
+  );
+
+  const newsData = news.map(({ uri, dateUpdated, title, date }) => {
```

Comment:
parse news posts separately, if they are less than 2 days old then Google News markup is added to indicate new news release